src/distributions/uniform.rs

```diff
@@ -287,18 +291,30 @@ pub trait UniformSampler: Sized {
     }
 }
 
-impl<X: SampleUniform> From<::core::ops::Range<X>> for Uniform<X> {
+impl<X: SampleUniform> From<Range<X>> for Uniform<X> {
     fn from(r: ::core::ops::Range<X>) -> Uniform<X> {
         Uniform::new(r.start, r.end)
     }
 }
 
-impl<X: SampleUniform> From<::core::ops::RangeInclusive<X>> for Uniform<X> {
+impl<X: SampleUniform> From<RangeInclusive<X>> for Uniform<X> {
     fn from(r: ::core::ops::RangeInclusive<X>) -> Uniform<X> {
         Uniform::new_inclusive(r.start(), r.end())
     }
 }
 
+impl<T: SampleUniform> Distribution<T> for Range<T> {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> T {
+        T::Sampler::sample_single(&self.start, &self.end, rng)
+    }
+}
+
+impl<T: SampleUniform> Distribution<T> for RangeInclusive<T> {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> T {
+        T::Sampler::sample_single_inclusive(self.start(), self.end(), rng)
+    }
+}
```

Comment:
Do we want this in addition to `gen_range(RANGE)`? To me it seems redundant and less desirable due to limited opportunities for documentation.